editoast/src/views/projects.rs

```diff
@@ -0,0 +1,177 @@
+use crate::error::Result;
+use crate::views::pagination::PaginationQueryParam;
+use crate::DbPool;
+
+use actix_web::dev::HttpServiceFactory;
+use actix_web::web::{self, block, scope, Data, Json, Path, Query};
+use actix_web::{delete, get, patch, post, HttpResponse, Responder};
+
+use crate::projects::ProjectData;
+use crate::projects::{Project, ProjectPatch};
+
+/// Returns `/projects` routes
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/projects")
+        .service((create, list))
+        .service(scope("/{project}").service((get, delete, patch)))
+}
+
+#[post("")]
+async fn create(db_pool: Data<DbPool>, data: Json<ProjectData>) -> Result<impl Responder> {
+    let project = block::<_, Result<_>>(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        Project::create(data.into_inner(), &mut conn)
+    })
+    .await
+    .unwrap()?;
+    Ok(HttpResponse::Created().json(project))
+}
+
+/// Return a list of projects
+#[get("")]
+async fn list(
+    db_pool: Data<DbPool>,
+    query_params: Query<PaginationQueryParam>,
+) -> Result<Json<Vec<Project>>> {
+    let page = query_params.page;
+    let per_page = query_params.page_size.unwrap_or(25).max(10);
+    block(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        Ok(Json(Project::list(&mut conn, page, per_page)))
+    })
+    .await
+    .unwrap()
+}
+
+/// Return a specific project
+#[get("")]
```

Comment:
Do you have to give the param here ?

Reply:
> Do you have to give the param here ?

No need, because I added the param when I declared the routes
.service(scope("/{project}").service((get, delete, patch))) (l.15)